pkg/gui/controllers/commits_files_controller.go

```diff
@@ -231,21 +235,47 @@ func (self *CommitFilesController) toggleForPatch(node *filetree.CommitFileNode)
 				}
 			}
 
-			// if there is any file that hasn't been fully added we'll fully add everything,
-			// otherwise we'll remove everything
-			adding := node.SomeFile(func(file *models.CommitFile) bool {
-				return self.c.Git().Patch.PatchBuilder.GetFileStatus(file.Name, self.context().GetRef().RefName()) != patch.WHOLE
+			selectedNodes = normalisedSelectedCommitFileNodes(selectedNodes)
+
+			nodesFullyInPatch, nodesPartiallyOrNotInPatch := utils.Partition(selectedNodes, func(node *filetree.CommitFileNode) bool {
```

Comment:
I would simplify this by updating AddFileWhole to make it do nothing if the file is already added whole. That way, we don't need to do this partition: we simply need to check if there is any node in the selection that contains a non-added or partially-added file, and if there is, we call AddFileWhole on each file, otherwise we call RemoveFile on each file. Note that RemoveFile doesn't need to change because it's already cheap to call for a file that's already been removed.

Reply:
Oh yeah that is a lot simpler! I think 509327c3822c09a41d934e749b09ca39284271e8 should cover this.